tests/faction_camp_test.cpp

```diff
@@ -0,0 +1,60 @@
+#include <cstddef>
+#include <iosfwd>
+#include <utility>
+#include <vector>
+
+#include "basecamp.h"
+#include "cata_catch.h"
+#include "clzones.h"
+#include "coordinates.h"
+#include "faction.h"
+#include "map.h"
+#include "map_helpers.h"
+#include "overmap.h"
+#include "overmapbuffer.h"
+#include "player_helpers.h"
+
+static const zone_type_id zone_type_CAMP_FOOD( "CAMP_FOOD" );
+static const zone_type_id zone_type_CAMP_STORAGE( "CAMP_STORAGE" );
+
+TEST_CASE( "camp_calorie_counting", "[camp]" )
+{
+    clear_avatar();
+    clear_map();
+    map &m = get_map();
+    const tripoint zone_loc = m.getabs( tripoint{ 5, 5, 0 } );
```

Comment:
This is an abs tripoint and needs to be typed as such,  tripoint_abs_ms.
If it had been,  the compiler could have told you about the errors on lines 38-40 abd 54-55 where you pass an abs coord to local methods.

Reply:
Hm I copied that wholesale from zones_custom_test.cpp, it seemed suspect to me but that hasn't been having any issues that I've seen.

https://github.com/CleverRaven/Cataclysm-DDA/blob/078a7b43be36c11f2431bf52f5f00d2d41c9a53a/tests/zones_custom_test.cpp#L14-L15

They seem to be doing it for the same reason - `mapgen_place_zone` only accepts raw tripoints and not typed points(!!!). (Not at my desktop right now so dumb question time) Is it possible to take our typed tripoint (tripoint_abs_ms/tripoint_bub_ms/etc) and turn it into a raw tripoint? There was some sticking point I can't recall, but it seems like something that should be easy.

Also judging by your messages in the dev discord's #technical-discussion channel I assume the chances to eoc_test.cpp should be reverted?